src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Data.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    public sealed partial class PlatformCompatabilityAnalyzer
+    {
+        private enum PlatformAttributeType
+        {
+            TargetPlatformAttribute,
+            MinimumOSPlatformAttribute,
+            ObsoletedInOSPlatformAttribute,
+            RemovedInOSPlatformAttribute,
+        }
+
+        private readonly struct PlatformAttributeInfo : IEquatable<PlatformAttributeInfo>
+        {
+            public PlatformAttributeType AttributeType { get; }
+            public string PlatformName { get; }
+            public Version Version { get; }
+
+            private PlatformAttributeInfo(PlatformAttributeType attributeType, string platformName, Version version)
+            {
+                AttributeType = attributeType;
+                PlatformName = platformName;
+                Version = version;
+            }
+
+            public static bool TryParsePlatformAttributeInfo(AttributeData osAttribute, out PlatformAttributeInfo parsedAttribute)
+            {
+                if (!osAttribute.ConstructorArguments.IsEmpty &&
+                    osAttribute.ConstructorArguments[0] is { } argument &&
```

Comment:
First time I see `is { }`. What does it do? Is it trying to cast it to an array of anything?

Reply:
No, it creates a new variable argument with the type and value of expression `osAttribute.ConstructorArguments[0]`, so a `TypedConstant` here. The expression returns true if the value is non-null. Its a fancy way of saying `var argument` with a null check.